opensquirrel/merger.py

```diff
@@ -59,8 +59,8 @@ def merge_single_qubit_gates(squirrel_ir: SquirrelIR):
     while statement_index < len(squirrel_ir.statements):
         statement = squirrel_ir.statements[statement_index]
 
-        if not isinstance(statement, Gate):
-            # Skip, since statement is not a gate
+        if not isinstance(statement, Gate) and not isinstance(statement,Measure):
```

Comment:
This can be done when the user inputs the code themselves. However, because of the decomposition step we add, the measurement statements get shifted up. Skipping immediately causes the logic of the decomposition to change. 

 i.e, this is what it looks like with the current implementation:
```
 qubit[3] qreg

Rz qreg[0], 1.5707963
X90 qreg[0]
Rz qreg[0], 2.9415927
X90 qreg[0]
Rz qreg[0], 3.1415927
CNOT qreg[1], qreg[0]
Rz qreg[0], -2.3521427
X90 qreg[0]
Rz qreg[0], 3.1415927
X90 qreg[0]
Rz qreg[0], 0.78945
CNOT qreg[1], qreg[0]
Rz qreg[2], 3.1415927
X90 qreg[2]
Rz qreg[2], 0.80159265
X90 qreg[2]
Rz qreg[1], -1.8561945
X90 qreg[1]
Rz qreg[1], 3.1415927
X90 qreg[1]
Rz qreg[1], 1.2853981
measure qreg[0]
measure qreg[2]
```

Adding the skip, changes it to this:
```
 qubit[3] qreg

Rz qreg[0], 1.5707963
X90 qreg[0]
Rz qreg[0], 2.9415927
X90 qreg[0]
Rz qreg[0], 3.1415927
CNOT qreg[1], qreg[0]
Rz qreg[0], -2.3521427
X90 qreg[0]
Rz qreg[0], 3.1415927
X90 qreg[0]
Rz qreg[0], 0.78945
CNOT qreg[1], qreg[0]
Rz qreg[1], -1.8561945
X90 qreg[1]
Rz qreg[1], 3.1415927
X90 qreg[1]
Rz qreg[1], 1.2853981
Rz qreg[2], 3.1415927
X90 qreg[2]
Rz qreg[2], 0.80159265
X90 qreg[2]
measure qreg[0]
measure qreg[2]
```
which does not match the other test cases for the decomposition

Reply:
Oke we can leave it in for now, but since there is nothing wrong with the second output (i.e. it is semantically identical to the first output), we should change it at a later stage. I'll make an issue for it.